### ISSUE TITLE
Allow logging metadata from `InitResourceContext`

### ISSUE THREAD

**Issue by danielgafni** [dagster-io/dagster#20102]
### What's the use case?

I have a bunch of resources which would really benefit from this. 

For instance, one of the resources provides a `RayCluster` on K8s, and another one is used to setup model training, and provides useful information like W&B tracking URL. 

The first one could automatically log information about the cluster (including compute costs which it's able to estimate internally), the second would log model training metadata. 

Right now I need to log all of this manually in every asset/op which is using these resources. 

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by cmpadden** [comment]
@danielgafni did you end up solving this?

The previous version of the `HexResource` was able to do this as it could `get_default_logger()` in the initializer.

https://github.com/dagster-io/community-integrations/blob/1106a5aa7d61b47689240ebb800e59da520efc66/libraries/dagster-hex/dagster_hex/resources.py#L43

However, as I refactor this to be a configurable resource, I'm unsure of the comparable functionality.

**Comment by danielgafni** [comment]
No, I believe I wasn't able to solve this issue. I also believe it's a considerable flaw in the current setup.

**Comment by danielgafni** [comment]
@cmpadden randomly came across this issue again... Looks like (1) you misunderstood my problem and (2) I misunderstood yours 🤣 

My problem was about logging **metadata**. Like `context.add_output_metadata`. 
Your problem was about accessing the logger. I think by now you probably already know it's accessible via `setup_for_execution`, `teardown_for_execution` or `yield_for_execution`.

**Comment by cmpadden** [comment]
> [@cmpadden](https://github.com/cmpadden) randomly came across this issue again... Looks like (1) you misunderstood my problem and (2) I misunderstood yours 🤣
> 
> My problem was about logging **metadata**. Like `context.add_output_metadata({"my_metric": value})`. Your problem was about accessing the logger. I think by now you probably already know it's accessible via `setup_for_execution`, `teardown_for_execution` or `yield_for_execution`.

Ah yeah, that was a total misunderstanding! Yup, I'm now familiar with those--thanks for following up.
